hermes-management/src/main/java/pl/allegro/tech/hermes/management/migration/maintainer/SupportTeamToMaintainerMigrator.java

```diff
@@ -0,0 +1,175 @@
+            );
+        } else {
+            subscriptionCounters.markSkipped(MAINTAINER_ALREADY_EXISTED_REASON);
+        }
+    }
+
+    private void migrateTopic(Topic topic, Group group, String sourceName, MaintainerExistsStrategy strategy, EntityMigrationCounters topicCounters) {
+        if (topic.getMaintainer() == null || strategy == MaintainerExistsStrategy.OVERRIDE) {
+            migrateEntity(topicCounters, "topic " + topic.getQualifiedName(),
+                    () -> topicService.updateTopic(topic.getName(), patchWithMaintainer(sourceName, group.getSupportTeam()), MIGRATION_USER)
+            );
+        } else {
+            topicCounters.markSkipped(MAINTAINER_ALREADY_EXISTED_REASON);
+        }
+    }
+
+    private void migrateEntity(EntityMigrationCounters topicCounters, String humanReadableEntityName, Runnable migration) {
+        try {
+            migration.run();
+            topicCounters.markMigrated();
+        } catch (Exception e) {
+            logger.info("Failed to migrate {}, skipping", humanReadableEntityName, e);
+            topicCounters.markSkipped(e.getClass().getTypeName());
+        }
+    }
+
+    private PatchData patchWithMaintainer(String sourceName, String supportTeam) {
+        return patchData().set("maintainer", new MaintainerDescriptor(sourceName, nullToEmpty(supportTeam))).build();
+    }
+
+    private static class EntityMigrationCounters {
+        private int migrated = 0;
+        private Map<String, Integer> skipped = new HashMap<>();
+
+        void markMigrated() {
+            migrated += 1;
+        }
+
+        void markSkipped(String reason) {
+            skipped.compute(reason, (key, old) -> (old != null ? old : 0) + 1);
+        }
+
+        EntityMigrationStats toStats() {
+            return new EntityMigrationStats(migrated, skipped);
+        }
+    }
+
+    public static class ExecutionStats {
+        private final EntityMigrationStats topics;
+        private final EntityMigrationStats subscriptions;
+
+        @JsonCreator
+        public ExecutionStats(@JsonProperty("topics") EntityMigrationStats topics,
+                              @JsonProperty("subscriptions") EntityMigrationStats subscriptions) {
+            this.topics = topics;
+            this.subscriptions = subscriptions;
+        }
+
+        @JsonProperty("topics")
+        public EntityMigrationStats topics() {
+            return topics;
+        }
+
+        @JsonProperty("subscriptions")
+        public EntityMigrationStats subscriptions() {
+            return subscriptions;
+        }
+
+        @Override
+        public String toString() {
+            return "ExecutionStats{" +
+                    "topics=" + topics +
+                    ", subscriptions=" + subscriptions +
+                    '}';
+        }
+    }
+
+    public static class EntityMigrationStats {
```

Comment:
I think it's best to leave it belonging to this particular migration for now. Since we don't have any more migrations at the moment I would be cautious of introducing general concepts prematurely. That's why I also didn't write any general migrations mechanism, just provided this particular migration, but via rather general HTTP API already.